basic_bootloader/src/bootloader/runner.rs

```diff
@@ -824,91 +812,57 @@ impl<'external, S: EthereumLikeTypes> Run<'_, 'external, S> {
-        self.system
-            .finish_global_frame((!deployment_success).then_some(&constructor_rollback_handle))?;
-
-        let _ = self.system.get_logger().write_fmt(format_args!(
-            "Return from constructor call, success = {}\n",
-            deployment_success
-        ));
-
-        resources_returned.reclaim(resources_for_deployer);
-        Ok(CompletedDeployment {
-            resources_returned,
-            deployment_result,
-        })
+        Ok((deployment_success, resources_returned, deployment_result))
     }
 }
 
 pub enum CallPreparationResult<'a, S: SystemTypes> {
     Success {
         next_ee_version: u8,
-        bytecode: &'a [u8],
-        code_version: u8,
-        unpadded_code_len: u32,
-        artifacts_len: u32,
-        actual_resources_to_pass: S::Resources,
         transfer_to_perform: Option<TransferInfo>,
-        resources_returned: S::Resources,
+        external_call_launch_params: ExecutionEnvironmentLaunchParams<'a, S>,
+        resources_in_caller_frame: S::Resources,
     },
     Failure {
-        resources_returned: S::Resources,
+        resources_in_caller_frame: S::Resources,
     },
 }
 
-/// Reads callee account and runs call preparation function
-/// from the system. Additionally, does token transfer if needed.
-fn run_call_preparation<'a, S: EthereumLikeTypes>(
-    is_entry_frame: bool,
+/// Read callee properties, execute additional checks, charge resources and perform additional EE-specific logic
+fn run_call_preparation<'a, S: EthereumLikeTypes, const IS_ENTRY_FRAME: bool>(
     system: &mut System<S>,
     ee_version: ExecutionEnvironmentType,
-    call_request: &ExternalCallRequest<S>,
+    mut call_request: ExternalCallRequest<'a, S>,
 ) -> Result<CallPreparationResult<'a, S>, BootloaderSubsystemError>
 where
     S::IO: IOSubsystemExt,
 {
-    let mut resources_available = call_request.available_resources.clone();
+    let mut resources_in_caller_frame = call_request.available_resources.take();
 
-    let r = if is_entry_frame {
+    let r = if IS_ENTRY_FRAME {
         // For entry frame we don't charge ergs for call preparation,
         // as this is included in the intrinsic cost.
-        resources_available.with_infinite_ergs(|inf_resources| {
+        resources_in_caller_frame.with_infinite_ergs(|inf_resources| {
             cycle_marker::wrap_with_resources!("prepare_for_call", inf_resources, {
-                prepare_for_call(
-                    system,
-                    ee_version,
-                    inf_resources,
-                    &call_request,
-                    is_entry_frame,
-                )
+                read_callee_account_properties(system, ee_version, inf_resources, &call_request)
             })
         })
     } else {
-        cycle_marker::wrap_with_resources!("prepare_for_call", resources_available, {
-            prepare_for_call(
+        cycle_marker::wrap_with_resources!("prepare_for_call", resources_in_caller_frame, {
```

Comment:
@antoniolocascio I'm not sure that it captures the scope that you intended anymore, pls take a look

Reply:
It doesn't, but that has been true for a while (since we move the transfer out of there). Can you just remove the cycle marker call? I can re-add it properly if I need to re-run this part of the native model, we don't use this marker for anything else